### ISSUE TITLE
flag not selected automatically while edit form

### ISSUE THREAD

**Issue by ShaileshInfyom** [gauravsoni119/ng2-tel-input#9]
i have used in reactive form method. code here
issue: when i open edit screen and patch a value into form. working fine. value displaying into input correctly but flag not changed. default flag appear. one i have change value then flag changed.
value after patch  for ex. +917878747474
```
<input type="text"   formControlName="value_field"
 ng2TelInput
[ng2TelInputOptions]="{initialCountry: 'us'}"
(hasError)="hasError($event)"
(ng2TelOutput)="hasOutPut($event)"
(countryChange)="onCountryChange($event)"
maxlength="45"/>
```
please help me how to achieve this and fixed my problem. thanks in advance.

**Comment by ShaileshInfyom** [comment]
any update?

**Comment by gauravsoni119** [comment]
You can get original intlTelInput object once this loaded. After this you can do something like below after editing :-
this.intlObj.intlTelInput("setCountry", "gb");

You can check the readme file to how to get intlTelInput instance.

**Comment by ShaileshInfyom** [comment]
Yes. I know but I haven't country code. just phone number available with country code. for eg. +917878747474

**Comment by gauravsoni119** [comment]
For me it's working, if I do like 

> **this.intlObj.intlTelInput("setNumber", "+917878747474");** 

then it will automatically change your flag. May be you can check your css file loaded properly or not.

**Comment by ketchup208** [comment]
> For me it's working, if I do like
> 
> > **this.intlObj.intlTelInput("setNumber", "+917878747474");**
> 
> then it will automatically change your flag. May be you can check your css file loaded properly or not.

SetNumber is not working for me ...

**Comment by mkpasala** [comment]
how you are getting this **this.intlObj** object in class method?

**Comment by akvaliya** [comment]
Any updates? I am facing the same issue. Flag is not changed for default value when it's filled from code.

**Comment by ketchup208** [comment]
> how you are getting this **this.intlObj** object in class method?


`(intlTelInputObject)="telInputObject($event)"` in the html
```
telInputObject(obj) {
    obj.intlTelInput('setCountry', this.country);
  }
```
in .ts file

**Comment by akilljain** [comment]
how to save the flag with the input field.

**Comment by akilljain** [comment]
> i have used in reactive form method. code here
> issue: when i open edit screen and patch a value into form. working fine. value displaying into input correctly but flag not changed. default flag appear. one i have change value then flag changed.
> value after patch for ex. +917878747474
> 
> ```
> <input type="text"   formControlName="value_field"
>  ng2TelInput
> [ng2TelInputOptions]="{initialCountry: 'us'}"
> (hasError)="hasError($event)"
> (ng2TelOutput)="hasOutPut($event)"
> (countryChange)="onCountryChange($event)"
> maxlength="45"/>
> ```
> 
> please help me how to achieve this and fixed my problem. thanks in advance.

can u solve it plz provided the solutions

**Comment by mevijaysaini** [comment]
Put **Ng2TelInputModule** after **FormsModule, ReactiveFormsModule** in app module****
